device_backends/DummyBackend/src/ExceptionDummyBackend.cc

```diff
@@ -103,26 +103,26 @@ namespace ChimeraTK {
 
   void ExceptionDummy::open() {
     if(throwExceptionOpen) {
-      thereHaveBeenExceptions = true;
-      throw(ChimeraTK::runtime_error("DummyException: open throws by request"));
+      throwExceptionCounter++;
+      setException("DummyException: open throws by request");
+      throw ChimeraTK::runtime_error("DummyException: open throws by request");
     }
     ChimeraTK::DummyBackend::open();
-    thereHaveBeenExceptions = false;
   }
 
   /********************************************************************************************************************/
 
   void ExceptionDummy::closeImpl() {
-    setException();
+    setException("Close ExceptionDummy");
```

Comment:
setException does not throw. I think this line is completely redundant, since the exception state cannot be seen from the outside. The backend is closed, hence any operation would lead to a logic_error. (Otherwise the spec is violated)

Reply:
```suggestion
```